core/src/math/matrix.cpp

```diff
@@ -1,58 +1,50 @@
 #include <jml/math/matrix.hpp>
+#include <internal/logger.hpp>
 
 #include <iostream>
 
 namespace jml {
 
-Matrix::Matrix(int m, int n) {
-    this->entries = new double(m * n);
-}
+Matrix::Matrix(int m, int n) { this->entries = new double(m * n); }
 
-Matrix::~Matrix() {
-    delete this->entries;
-}
+Matrix::~Matrix() { delete this->entries; }
 
-int Matrix::get_n_rows() {
-    return m;
-}
+int Matrix::get_position(int i, int j) { return i * this->n + j; }
 
-int Matrix::get_n_cols() {
-    return n;
-}
+int Matrix::get_n_rows() { return m; }
 
-int Matrix::get_position(int i, int j) {
-    return i * this->n + j;
-}
+int Matrix::get_n_cols() { return n; }
 
 void Matrix::set_entry(int i, int j, double value) {
 
     if (i < 0 || i >= this->m) {
-        std::cout << "Row number " << i << " out of bounds." << std::endl;
+        LOGGER->log(Log() << WARN << "Row number " << i << " out of bounds.\n");
     }
     if (j < 0 || j >= this->n) {
-        std::cout << "Column number " << i << " out of bounds." << std::endl;
+        LOGGER->log(Log() << WARN << "Column number " << i
+                          << " out of bounds.\n");
     }
 
     this->entries[get_position(i, j)] = value;
-
 }
 
 std::unique_ptr<Vector> Matrix::multiply(std::unique_ptr<Vector> in) {
 
     if (in->get_size() != this->n) {
-        std::cout << "Tried to multiply a vector of size " << in->get_size()
-        << " with a matrix containing " << this->n << " columns." << std::endl;
+        LOGGER->log(Log(WARN) << "Tried to multiply a vector of size "
```

Comment:
Is this a warning? It seems like an error. (Also with the one for vectors below). What do you think?